src/services/chatroom.service.ts

```diff
@@ -0,0 +1,21 @@
+import { CreateChatroomDto } from '@/dtos';
+import { Chatroom } from '@/entities/nosql';
+import { HttpException } from '@/exceptions';
+import { model } from 'mongoose';
+
+class ChatroomService {
+  public async createChatroom({ userId, receiverId, isGroup }: CreateChatroomDto): Promise<void> {
+    const chatroom = model('Chatroom', Chatroom);
+    const createChat = new chatroom({
+      isGroup: isGroup,
+      users: [userId, receiverId],
+      message: [],
+      createdDate: Date.now(),
+    });
+    const checkExist = await chatroom.exists({ users: createChat.users });
```

Comment:
Shouldn't we check for if the chatroom exists before we create one?
so line 15 and 16 would be at the top.

Reply:
True, I did create the object. But I didn't save it until after checking whether it exists. I changed it though